docs/rpc-endpoints.md

```diff
@@ -0,0 +1,255 @@
+# RPC Endpoints
+
+### POST /v2/transactions
+
+This endpoint is for posting _raw_ transaction data to the node's mempool.
+
+Rejections result in a 400 error.
+
+### GET /v2/accounts/[Principal]
+
+Get the account data for the provided principal.
+The principal string is either a Stacks address or a Contract identifier (e.g., 
+`SP31DA6FTSJX2WGTZ69SFY11BH51NZMB0ZW97B5P0.get-info`
+
+Returns JSON data in the form:
+
+```
+{
+ "balance": 100,
+ "nonce": 1,
+ "balanceProof": "01fa...",
+ "nonceProof": "01ab...",
+}
+```
+
+Where balance (a u128) and nonce (a u64) are both integers, and the proofs are provided as
+hex strings.
+
+For non-existent accounts, this _does not_ 404, rather it returns an
+object with balance and nonce of 0.
+
+This endpoint also accepts a querystring parameter `?proof=` which when supplied `0`, will return the
+JSON object _without_ the `balanceProof` or `nonceProof` fields.
+
+### POST /v2/map_entry/[Stacks Address]/[Contract Name]/[Map Name]
+
+Attempt to fetch data from a contract data map. The contract is identified with [Stacks Address] and
+ [Contract Name] in the URL path. The map is identified with [Map Name].
+ 
+The _key_ to lookup in the map is supplied via the POST body. This should be supplied as the hex string
+serialization of the key (which should be a Clarity value). Note, this is a _JSON_ string atom.
+
+Returns JSON data in the form:
+
+```
+{
+ "data": "01ce...",
+ "marfProof": "01ab...",
+}
```

Comment:
What's our take on enveloping the responses from all our endpoints with a structure such as:
```
{
  success: true,
  data: { ... }
}
```
and 
```
{
  success: false,
  error: "Map not found" / "Contract not found" / ...,
}
```

Reply:
I don't think we have enough information yet about whether or not this is a good idea.  I could be convinced that there are some API endpoints where the `success` and/or `data` keys are unnecessary and would only add confusion.

Reply:
That's fine with me.

Reply:
Thanks!